src/iceberg/result.h

```diff
@@ -86,4 +93,67 @@ auto InvalidExpressionError(const std::format_string<Args...> fmt, Args&&... arg
                             .message = std::format(fmt, std::forward<Args>(args)...)});
 }
 
+/// \brief Create an unexpected error with kInvalidSchema
```

Comment:
Perhaps we can use a macro to define these to avoid duplicates?

Reply:
done, thanks